test/browser/legacy_full_page_refresh_test.rb

```diff
@@ -38,28 +54,34 @@ class LegacyPagesFullPageRefreshTest < ActionDispatch::IntegrationTest
     refute page.has_selector?("[refresh-never]")
   end
 
-  test "going to a URL that will error 500, and hitting the browser back button, we see the correct page (and not the 500)" do
+  test "going to a URL that should error 500" do
     click_link "I will throw an error 500"
-    has_text = false
-    while !has_text
-      has_text = page.assert_text('Error 500!')
-      sleep 1
-    end
+    wait_until_page_has("Error 500!")
+
     assert_not_equal "Sample Turbograft Application", page.title
+  end
+
+  test "going to a URL that will error 500, and hitting the browser back button, we see the correct page (and not the 500)" do
+    click_link "I will throw an error 500"
+    wait_until_page_has("Error 500!")
```

Comment:
Awesome, I'll try to port it over to use this instead although I assume people saw that it was flaky and decided to implement the `while` loop approach leading to even more flaky tests.

I'll actually move this another PR.